connectors/idp-extraction/src/main/java/io/camunda/connector/idp/extraction/service/StructuredService.java

```diff
@@ -126,21 +127,27 @@ private StructuredExtractionResult processExtractedData(
               }
               Object confidenceScore = response.confidenceScore().get(key);
 
-              // if the list is empty will not filter any fields.
-              if ((input.includedFields() == null
-                      || input.includedFields().isEmpty()
-                      || input.includedFields().contains(key))
-                  && value != null) {
+              // Check if field should be included based on includedFields filter
+              boolean shouldInclude;
+              if (input.includedFields() == null || input.includedFields().isEmpty()) {
+                shouldInclude = true;
+              } else {
+                shouldInclude = input.includedFields().contains(key);
+              }
+
+              // Include the field if it should be included and value is not null
+              if (shouldInclude && value != null) {
```

Comment:
🔧 Nit: could be extracted to a method `shouldIncludeField(includedFields, key, value)` or similar for better readability.